crates/node/api/src/node.rs

```diff
@@ -26,6 +27,25 @@ pub trait NodeTypes: Send + Sync + Unpin + 'static {
     type Engine: EngineTypes;
 }
 
+/// A [`NodeTypes`] type builder
+#[derive(Debug)]
+pub struct AnyNodeTypes<P = (), E = ()>(PhantomData<P>, PhantomData<E>);
```

Comment:
here we want

`impl default for AnyNodeTypes`

and two builder style functions that can be used to set the types, like

```rust

fn engine<T>(self) -> AnyNodeTypes<P, T>

```